File: test_cases/python/claude/17.py
def run_tests(solution_module):
    f = solution_module.minCut
    # single char — already palindrome
    assert f('a') == 0
    # already a palindrome
    assert f('aba') == 0
    # two different chars
    assert f('ab') == 1
    # classic example
    assert f('aab') == 1
    # all same chars — zero cuts
    assert f('aaaa') == 0

File: test_cases/c/claude/17.c
#include <assert.h>
#include <stdio.h>

int minCut(char* s);

int main(void) {
    assert(minCut("aab") == 1);
    assert(minCut("a") == 0);
    assert(minCut("ab") == 1);
    assert(minCut("aaaa") == 0);
    assert(minCut("abcde") == 4);
    return 0;
}